script/foundry/utils/DeployHelper.sol

```diff
@@ -300,17 +303,39 @@ contract DeployHelper is Script, BroadcastManager, JsonDeploymentHandler, Storag
                 address(royaltyModule),
                 address(licenseRegistry),
                 address(disputeModule),
-                address(licenseToken)
+                licenseTokenAddr
```

Comment:
The changes in the contracts are correct, passing the values from storage to immutable and initializing in the constructors.

Here we use CREATE3 to deploy and calculate the implementations and the proxy contracts, that's good.

However, we have to use always the proxy address to reference an upgradeable contract from another, if we pass the implementation address around, we will use the state of the implementations, not the proxy. Upgrading a proxy would break the protocol, since the new implementation will lose all the previous state.

I believe `licenseTokenAddr` is the deterministic address of the implementation for LicenseToken, not the proxy? We need the proxy address.

I was workshopping a similar approach to this, where we would
- deploy the proxies first with an empty implementation to get the addresses
- then we can deploy the implementation contracts with the relevant proxy addresses in the constructor
- then we upgrade the proxies to the proper implementation
https://github.com/storyprotocol/protocol-core-v1/pull/125/files

It's not as efficient, sharing in case it helps to visualize the issue. If we can use create3 to predict the proxy and the implementations accurately (as your PR intends to), that would be best.

Reply:
The licenseTokenAddr actually serves as a proxy address. This is because the proxy is deployed using CREATE3 in the PR. I have added a check to confirm that the deployed address is indeed a proxy, and it correctly references the intended implementation address.

```solidity
require(_loadProxyImpl(address(licenseRegistry)) == impl, "LicenseRegistry Proxy Implementation Mismatch");
```

Reply:
Thanks, what confused me is that 
- the salt is created with the implementation code
- the deployed contract is actually the proxy, using the bytecode of the proxy, but the salt of the implementation code.
- And that works because CREATE3 only depends on the salt.

Using the bytecode to generate the salt made me believe it was dependent to the deployed bytecode for some reason, but they are not.

We could have just hashed contract name + timestamp or something and it would work the same.

Looking good then!